File: lib/models/ozel_gosterge_data.dart
class OzelGostergeData {
  final String gostergeName;
  final List<double> dailyValues;
  final List<String> dates;
  final List<double> monthlyChanges;
  final List<String> monthlyDates;

  OzelGostergeData({
    required this.gostergeName,
    required this.dailyValues,
    required this.dates,
    required this.monthlyChanges,
    required this.monthlyDates,
  });

  // Yılbaşından itibaren değişim hesapla (son değer - 100)
  double getYearToDateChange() {
    if (dailyValues.isEmpty) return 0.0;
    return dailyValues.last - 100.0;
  }

  // En son aylık değişim oranını al
  double getLatestMonthlyChange() {
    if (monthlyChanges.isEmpty) return 0.0;
    return monthlyChanges.last;
  }
}

File: src/models/OzelGostergeData.ts
export interface OzelGostergeData {
  gostergeName: string;
  dailyValues: number[];
  dates: string[];
  monthlyChanges: number[];
  monthlyDates: string[];
}

export class OzelGostergeDataModel implements OzelGostergeData {
  gostergeName: string;
  dailyValues: number[];
  dates: string[];
  monthlyChanges: number[];
  monthlyDates: string[];

  constructor(
    gostergeName: string,
    dailyValues: number[],
    dates: string[],
    monthlyChanges: number[],
    monthlyDates: string[]
  ) {
    this.gostergeName = gostergeName;
    this.dailyValues = dailyValues;
    this.dates = dates;
    this.monthlyChanges = monthlyChanges;
    this.monthlyDates = monthlyDates;
  }

  getYearToDateChange(): number {
    if (this.dailyValues.length === 0) return 0.0;
    return this.dailyValues[this.dailyValues.length - 1] - 100.0;
  }

  getLatestMonthlyChange(): number {
    if (this.monthlyChanges.length === 0) return 0.0;
    return this.monthlyChanges[this.monthlyChanges.length - 1];
  }
}

